### ISSUE TITLE
public-api reports a relative import inside an @x file as a public API violation

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#40]
## Problem

A slice's `@x` folder is the standard FSD way to expose a public API for cross-imports. A relative import inside such a file points at the slice's own code, but the `public-api` rule reports it and suggests the path `".."`.

## Reproduction

`src/entities/foo/@x/bar.ts`:

```ts
export { thing } from '../model/thing';
```

```
1:23  error  Absolute imports are only allowed from public api ("..")  @conarti/feature-sliced/public-api  [should-be-from-public-api]
```

The same statement in `src/entities/foo/index.ts` (`export { thing } from './model/thing'`) produces no report, which is the expected behaviour.

## Expected

A relative import inside a slice, including one written in an `@x` file, is not a public API violation.

## Impact

Default preset, no options. Measured on a public project of 2277 files: 65 of its 73 `public-api` reports come from `@x` files.

Related: #31 fixed the `unknown-segment` message for `@x` paths in d388bc9. This is the other message of the same rule.

## Plugin version

`@conarti/eslint-plugin-feature-sliced@2.0.0`
